lib/recognizer/accounts/user.ex

```diff
@@ -117,7 +117,15 @@ defmodule Recognizer.Accounts.User do
     |> validate_format(:email, ~r/^[^\s]+@[^\s]+$/, message: "must have the @ sign and no spaces")
     |> validate_length(:email, max: 160)
     |> unsafe_validate_unique(:email, Repo)
-    |> unique_constraint(:email)
+    |> maybe_downcase_email()
+  end
+
+  defp maybe_downcase_email(changeset) do
+    if email = get_change(changeset, :email) do
+      put_change(changeset, :email, String.downcase(email))
+    else
+      changeset
+    end
```

Comment:
Why do we need to maybe downcase here? If it's already downcased it should a no-op, and after the database migration, everything should already be lower cased.

```suggestion
    |> update_change(:email, &String.downcase/1)
```

Reply:
It needs to be downcased anytime we save it otherwise the user can update it post-migration to include Capitalized letters via the update changeset.